src/Features/CSharp/Portable/Completion/KeywordRecommenders/FieldKeywordRecommender.cs

```diff
@@ -20,11 +22,36 @@ internal class FieldKeywordRecommender : AbstractSyntacticSingleKeywordRecommend
         SyntaxKind.EnumDeclaration,
     };
 
-    public FieldKeywordRecommender()
-        : base(SyntaxKind.FieldKeyword)
+    protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
     {
+        // `[field:` is legal in an attribute within a type.
+        if (context.IsMemberAttributeContext(s_validTypeDeclarations, cancellationToken))
+            return true;
+
+        // Check if we're within a property accessor where the `field` keyword is legal.  Note: we do not do a lang
+        // version check here.  We do not want to interfere with users trying to use/learn this feature.  The user will
+        // get a clear message if they're not on the right lang version telling them about the issue, and offering to
+        // upgrade their project if they way.
+        if (context.IsAnyExpressionContext || context.IsStatementContext)
```

Comment:
💭 Is there a case where `IsAnyExpressionContext` is false but `IsStatementContext` is true?